src/charm.py

```diff
@@ -0,0 +1,107 @@
+        super().__init__(*args, plugin="route53")
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+
+    @property
+    def _aws_access_key_id(self) -> str:
+        """Returns aws access key from config."""
+        return self.model.config.get("aws_access_key_id")
+
+    @property
+    def _aws_hosted_zone_id(self) -> str:
+        """Returns aws hosted zone id from config."""
+        return self.model.config.get("aws_hosted_zone_id")
+
+    @property
+    def _aws_secret_access_key(self) -> str:
+        """Returns aws secret access key from config."""
+        return self.model.config.get("aws_secret_access_key")
+
+    @property
+    def _aws_region(self) -> str:
+        """Returns aws region from config."""
+        return self.model.config.get("aws_region")
+
+    @property
+    def _aws_max_retries(self) -> str:
+        """Returns aws max retries from config."""
+        return self.model.config.get("aws_max_retries")
+
+    @property
+    def _aws_polling_interval(self) -> str:
+        """Returns aws polling interval from config."""
+        return self.model.config.get("aws_polling_interval")
+
+    @property
+    def _aws_propagation_timeout(self) -> str:
+        """Returns aws propagation timeout from config."""
+        return self.model.config.get("aws_propagation_timeout")
+
+    @property
+    def _aws_ttl(self) -> str:
+        """Returns aws ttl from config."""
+        return self.model.config.get("aws_ttl")
+
+    @property
+    def _plugin_config(self) -> Dict[str, str]:
+        """Plugin specific additional configuration for the command."""
+        additional_config = {
+            "AWS_REGION": self._aws_region,
+            "AWS_HOSTED_ZONE_ID": self._aws_hosted_zone_id,
+            "AWS_ACCESS_KEY_ID": self._aws_access_key_id,
+            "AWS_SECRET_ACCESS_KEY": self._aws_secret_access_key,
+        }
+        if self._aws_max_retries:
+            additional_config["AWS_MAX_RETRIES"] = self._aws_max_retries
+        if self._aws_polling_interval:
+            additional_config["AWS_POLLING_INTERVAL"] = self._aws_polling_interval
+        if self._aws_propagation_timeout:
+            additional_config["AWS_PROPAGATION_TIMEOUT"] = self._aws_propagation_timeout
+        if self._aws_ttl:
+            additional_config["AWS_TTL"] = self._aws_ttl
+        return additional_config
+
+    def _on_config_changed(self, _):
+        """Handles config-changed events."""
+        if not self._aws_access_key_id or not self._aws_secret_access_key:
+            self.unit.status = BlockedStatus(
+                "aws-access-key-id, aws-secret-access-key must be set."
+            )
+            return
+        if not self._aws_region:
+            self.unit.status = BlockedStatus("aws-region must be set.")
+            return
+        if not self._aws_hosted_zone_id:
+            self.unit.status = BlockedStatus("aws-hosted-zone-id must be set.")
+            return
```

Comment:
Is there any reason for handling the params separately?

Reply:
only the message, should I merge them into one?

Reply:
I think I'd do something similar to what we have done in the `charmed-magma-orchestrator` for the relations. [Click me](https://github.com/canonical/charmed-magma-orchestrator/blob/main/orc8r-orchestrator-operator/src/charm.py#L233)